### PR TITLE
gh actions: e2e: add basic-install lane

### PR DESCRIPTION
**Pull request by ffromani**
This PR is the first step towards fully automated CI.
First and foremost, we add the very first, coarse grained, RBAC definitions.
Then we kick off a kubernetes e2e lane which run only the bare minimum tests.
All the work in this PR is meant to be further refined in followup PRs.

Next steps:

- add more tests to the install lane
- run the imported RTE e2e suite
- make platform autodetection work
- run the lane(s) also against OCP

### CODE REVIEW

File: hack/kind-config-e2e-no-registry.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+kubeadmConfigPatches:
+- |
+  kind: KubeletConfiguration
+  cpuManagerPolicy: "static"
```
Comment (Tal-or) [inline hack/kind-config-e2e-no-registry.yaml:6]: What about ```memoryManagerPolicy``` and ```reservedMemory```?

Comment (fromanirh) [reply hack/kind-config-e2e-no-registry.yaml:6]: TBD :)

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0 # Fetch all history for all tags and branches
+
+    - name: Set up golang
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16
```
Comment (cynepco3hahue) [inline .github/workflows/e2e.yml:31]: please use 1.17, to be aligned with k8s

File: .github/workflows/e2e.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      run: |
+        make binary-e2e
+
+    - name: Build image
+      run: |
+        docker build . -t ${built_image}
+
+    - name: Create K8s Kind Cluster
+      run: |
+        kind create cluster --config=hack/kind-config-e2e-no-registry.yaml --image kindest/node:v1.21.1@sha256:69860bda5563ac81e3c0057d654b5253219618a22ec3a346306239bba8cfa1a6
```
Comment (cynepco3hahue) [inline .github/workflows/e2e.yml:46]: from where is coming kind binary? Can we use a newer version for the kindest/node?

**Review by cynepco3hahue — COMMENTED** [review]
I am very curious what machines do we get for github CI, and does it have some limitations on runs per period of time?

File: .github/workflows/e2e.yml (hunk 2 above)
Comment (fromanirh) [reply .github/workflows/e2e.yml:31]: done

File: .github/workflows/e2e.yml (hunk 3 above)
Comment (fromanirh) [reply .github/workflows/e2e.yml:46]: It's part of the base image (added comment to act as reminder).
Kind still recommends this image: https://github.com/kubernetes-sigs/kind/releases/tag/v0.11.1
once the lane is stable I'd like to try the 1.22 image.

**Comment by fromanirh** [comment]
re: gh CI machines are simple 1-cpu machines. We have a time budget but never hit the limit yet.

**Comment by cynepco3hahue** [comment]
> re: gh CI machines are simple 1-cpu machines. We have a time budget but never hit the limit yet.

@fromanirh Do you know if it is per repo time limit or per organization?

**Comment by ffromani** [comment]
> > re: gh CI machines are simple 1-cpu machines. We have a time budget but never hit the limit yet.
> 
> @fromanirh Do you know if it is per repo time limit or per organization?

no idea. We will need to look into the gh docs.

**Pull request merged**